src/main/webapp/scripts/ua_kuali_application.js

```diff
@@ -55,6 +55,7 @@ jQuery( document ).ready(function() {
  */
 function loadLeadUnitName(unitNumberFieldName, leadUnitNameFieldName ) {
 	var unitNumber = dwr.util.getValue( unitNumberFieldName );
+	clearRecipients( leadUnitNameFieldName, "" );
 
 	if (unitNumber=='') {
```

Comment:
Since we're already clearingRecipients at the start of the method, this part of the if can be refactored:
if (unitNumber && unitNumber.trim().length > 0 ){
   var dwrReply = {.....